domain/domain.js

```diff
@@ -0,0 +1,45 @@
+const ARR_LAST_IDX = -1
+
+export function evaluateGuess(answer, randomNum, inputValue, max) {
+  const lastGuess = answer.at(ARR_LAST_IDX);
+
+  if (lastGuess < randomNum && inputValue <= max) return "UP"
+  if (lastGuess > randomNum && inputValue <= max) return "DOWN"
+  if (lastGuess === randomNum && inputValue <= max) return "CORRECT"
+}
+
+export function validateMinMax(input) {
+  const parts = input.split(",")
+  
+    if (parts.length === 2 && parts.every((part) => part !== "" && Number(part)) && Number(parts[0]) < Number(parts[1])) {
+      const min = Number(input.split(",")[0]?.trim())
+      const max = Number(input.split(",")[1]?.trim())
+  
+      return {min, max}
+    }
+  
+    return null
+  }
+  
+  export function validateCount(input) {
+    if ( input !== "" && Number(input) > 0 && !isNaN(input) ){
+      return Number(input)
+    }
+  
+    return null
+  }
+  
+  export function validateInput(input,answer, max) {
+    if (
+      input > 0 &&
+      input <= max &&
+      !answer.includes(Number(input)) && 
+      input !== "" && 
+      input !== null && 
+      input !== undefined
+    ) {
```

Comment:
P5;

저는 논리 연산자가 여러개가 얽혀있을 때 기대 결과값이 한눈에 안들어와 시간도 많이 잡아먹고, 실수하는 경우가 많이 있더라구요.

이런 경우에는 코드가 길어지더라도 맥락에 맞게 나누어 봐도 좋을 것 같아요!

```suggestion
    if (input === '' || input === null || input === undefined) return;
    
    if (input < min) return;
    
    if (input > max) return;
    
    if (!answer.includes(Number(input))) return;
    
    return Number(input);
```

Reply:
코드 가독성 생각해서 로직 구현하겠습니다. and 연산자로 작성할때도 저도 약간 헷갈려 했던게 생각났습니다.